libdnf/transaction/transaction_item_reason.cpp

```diff
@@ -42,12 +42,14 @@ std::string TransactionItemReason_to_string(TransactionItemReason reason) {
     return "";
 }
 
-
+// TransactionItemReason::UNKNOWN will have higher value than TransactionItemReason::DEPENDENCY
+// Important for autoremove to prevent unwanted removal
+// https://bugzilla.redhat.com/show_bug.cgi?id=1921063
 static TransactionItemReason order[] = {
-    TransactionItemReason::UNKNOWN,
     TransactionItemReason::CLEAN,
     TransactionItemReason::WEAK_DEPENDENCY,
     TransactionItemReason::DEPENDENCY,
+    TransactionItemReason::UNKNOWN,
```

Comment:
This looks like an obviously bad design. The comment above doesn't really describe the problem and the linked bug contains a lot of noise, requiring anyone trying to understand this to spend a lot of time filtering it out and analyzing the situation. The description here should be clean and complete.

Besides, I'm reasonably sure we don't want this and it won't be needed if we store the system state reasons separately. But then I may not fully understand the issue (because, again, it's just not well documented here).